### PR TITLE
tests: Fix NonCoherentMemoryMapping test

### PR DESCRIPTION
**Pull request by mark-lunarg**
New validation checks caused failures in this older LVT on the Pixel-C under Android.  Updated to use better Vulkan.

@cnorthrop, can you see if this fixes the Pixel issues?

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -16709,12 +16708,12 @@ TEST_F(VkPositiveLayerTest, NonCoherentMemoryMapping) {
     // Map with prime offset and size
     // Flush/Invalidate subrange of mapped area with prime offset and size
     m_errorMonitor->ExpectSuccess();
-    err = vkMapMemory(m_device->device(), mem, allocation_size - 137, 109, 0, (void **)&pData);
+    err = vkMapMemory(m_device->device(), mem, 11 * atom_size, 9 * atom_size, 0, (void **)&pData);
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: This range is too big, assuming the worst case noncoherent atom size of 256. The buffer is 0x1000; the end of this mapping is at 0x1400.

**Comment by cnorthrop** [comment]
PixelC still gets the original error.

I also getting the following on a number of platforms:

[ RUN      ] VkPositiveLayerTest.NonCoherentMemoryMapping
Unexpected: vkFlushMappedMemoryRanges: Flush/Invalidate size or offset (4092, 4089) exceed the Memory Object's upper-bound (20). For more information refer to Vulkan Spec Section '10.2.1. Host Access to Device Memory Objects' which states 'If size is not equal to VK_WHOLE_SIZE, offset and size must specify a range contained within the currently mapped range of memory' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#VkMappedMemoryRange)
/home/buildbot/LVL-buildbot/LVL-ubuntu-slave/dev_ubuntu_android/build/build-android/jni/../..//tests/layer_validation_tests.cpp:16718: Failure
Value of: err
  Actual: -1000011001
Expected: VK_SUCCESS
Which is: 0
UNKNOWN_RESULT
ObjectTracker(ERROR): object: 0x2 type: 8 location: 347 msgCode: 3: OBJ ERROR : For device 0xef8c1640, Device Memory object 0x2 has not been destroyed.
MEM(ERROR): object: 0x2 type: 8 location: 4518 msgCode: 7: Mem Object 0x2 has not been freed. You should clean up this memory by calling vkFreeMemory(0x2) prior to vkDestroyDevice().
[  FAILED  ] VkPositiveLayerTest.NonCoherentMemoryMapping (29 ms)

**Comment by mark-lunarg** [comment]
@cnorthrop, can you give this set a shot?  It worked on our android system here but the local farm is wonky today.

**Comment by cnorthrop** [comment]
Everything but PixelC is cleared up now.  Still seeing the original failure though:

[ RUN      ] VkPositiveLayerTest.NonCoherentMemoryMapping
Unexpected: vkFlushMappedMemoryRanges: Offset in pMemRanges[0] is 0xd, which is not a multiple of VkPhysicalDeviceLimits::nonCoherentAtomSize (0x40). For more information refer to Vulkan Spec Section '10.2.1. Host Access to Device Memory Objects' which states 'offset must be a multiple of VkPhysicalDeviceLimits::nonCoherentAtomSize' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#VkMappedMemoryRange)
Unexpected: vkFlushMappedMemoryRanges: Flush/Invalidate offset (13) is less than Memory Object's offset (320). For more information refer to Vulkan Spec Section '10.2.1. Host Access to Device Memory Objects' which states 'If size is equal to VK_WHOLE_SIZE, offset must be within the currently mapped range of memory' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#VkMappedMemoryRange)
/home/buildbot/LVL-buildbot/LVL-ubuntu-slave/dev_ubuntu_android/build/build-android/jni/../..//tests/layer_validation_tests.cpp:16702: Failure
Value of: err
  Actual: -1000011001
Expected: VK_SUCCESS
Which is: 0
UNKNOWN_RESULT
ObjectTracker(ERROR): object: 0x2 type: 8 location: 347 msgCode: 3: OBJ ERROR : For device 0x7c247ad000, Device Memory object 0x2 has not been destroyed.
MEM(ERROR): object: 0x2 type: 8 location: 4518 msgCode: 7: Mem Object 0x2 has not been freed. You should clean up this memory by calling vkFreeMemory(0x2) prior to vkDestroyDevice().
[  FAILED  ] VkPositiveLayerTest.NonCoherentMemoryMapping (819 ms)

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -16690,10 +16690,9 @@ TEST_F(VkPositiveLayerTest, NonCoherentMemoryMapping) {
...
 
-    // Map/Flush/Invalidate using WHOLE_SIZE and a prime offset and entire
-    // mapped range
+    // Map/Flush/Invalidate using WHOLE_SIZE and an offset and entire mapped range
     m_errorMonitor->ExpectSuccess();
-    err = vkMapMemory(m_device->device(), mem, 13, VK_WHOLE_SIZE, 0, (void **)&pData);
+    err = vkMapMemory(m_device->device(), mem, 5 * atom_size, VK_WHOLE_SIZE, 0, (void **)&pData);
     ASSERT_VK_SUCCESS(err);
     mmr.sType = VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE;
     mmr.memory = mem;
```
Comment (cnorthrop) [inline tests/layer_validation_tests.cpp]: I think this is the problem:

    mmr.offset = 13;

**Comment by mark-lunarg** [comment]
@cnorthrop, I feel good about this one -- totally missed the 13.  Also updated size to a multiple of atom_size as Chris Suggested.

**Comment by cnorthrop** [comment]
That worked - thanks Mark!

**Comment by mark-lunarg** [comment]
Pushed to Master.

**Pull request closed without merging**